components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -96,11 +98,32 @@ BraveReferralsService::BraveReferralsService(PrefService* pref_service,
       api_key_(api_key),
       platform_(platform),
       weak_factory_(this) {
+  registrar_.Add(this, chrome::NOTIFICATION_PROFILE_ADDED,
+                 content::NotificationService::AllBrowserContextsAndSources());
 }
 
 BraveReferralsService::~BraveReferralsService() {
 }
 
+void BraveReferralsService::Observe(
+    int type,
+    const content::NotificationSource& source,
+    const content::NotificationDetails& details) {
+  switch (type) {
+    case chrome::NOTIFICATION_PROFILE_ADDED: {
```

Comment:
Why don't we use the observer? see the notification doc
`  // Use ProfileManagerObserver::OnProfileAdded instead of this notification.`

Reply:
yea, I suppose this class already has a ton of layering violations so pulling in ProfileManager is no worse than pulling in the notification. We really need to move this out of components